error[E0034]: multiple applicable items in scope
  --> $DIR/trait-alias-ambiguous.rs:21:7
   |
LL |     t.foo();
   |       ^^^ multiple `foo` found
   |
note: candidate #1 is defined in an impl of the trait `inner::A` for the type `u8`
  --> $DIR/trait-alias-ambiguous.rs:8:9
   |
LL |         fn foo(&self) {}
   |         ^^^^^^^^^^^^^
   = help: to disambiguate the method call, write `inner::A::foo(t)` instead
note: candidate #2 is defined in an impl of the trait `inner::B` for the type `u8`
  --> $DIR/trait-alias-ambiguous.rs:11:9
   |
LL |         fn foo(&self) {}
   |         ^^^^^^^^^^^^^
   = help: to disambiguate the method call, write `inner::B::foo(t)` instead

error: aborting due to previous error

For more information about this error, try `rustc --explain E0034`.
